src/main/python/general_analysis/simulation_health_analysis.py

```diff
@@ -0,0 +1,47 @@
+import os
+import sys
+from glob import glob
+
+
+detectors = {
+                "deadLetter": lambda line: "DeadLetter" in line,
+                "actor died" : lambda line: "terminated unexpectedly" in line,
+                "warn": lambda line: " WARN " in line, 
+                "error": lambda line: " ERROR " in line, 
+                "stacktrace": lambda line: line.startswith("\tat ")   
```

Comment:
I now see how you got around it. I think there should be a comment as to why it is handled separately from other items to make it so that other maintainers don't need to follow the logic to understand it. Also, there is a bug that probably isn't worth handling - if the last line of the file is a stack trace then you will be one short